doc/specs/#2046 - Command Palette.md

```diff
@@ -248,6 +290,77 @@ For example, consider the following list of commands:
+we're successful in parsing the commandline, we can close the palette and
+dispatch the commandline. If the commandline had errors, we should reveal a text
+box with an error message below the text input. We'll leave the palette open
+with their entered command, so they can edit the commandline and try again. We
+should _probably_ leave the message up for a few seconds once they've begun
+editing the commandline, but eventually hide the message (ideally with a motion
+animation).
+
+### Switching Between Modes
+
+**TODO**: This is a topic for _discussion_.
+
+How do we differentiate Action Mode from Commandline Mode?
+
+I think there should be a character that the user types that switches the mode.
+This is reminiscent of how the command palette works in applications like VsCode
+and Sublime Text. The same UI is used for a number of functions. In the case of
+VsCode, when the user opens the palette, it's initially in a "navigate to file"
+mode. When the user types the prefix character `@`, the menu seamlessly switches
+to a "navigate to symbol mode". Similarly, users can use `:` for "go to line"
+and `>` enters an "editor command" mode.
+
+I believe we should use a similarly implemented UI. The UI would be in one of
+the two modes by default, and typing the prefix character would enter the other
+mode. If the user deletes the prefix character, then we'd switch back into the
+default mode.
+
+When the user is in Action Mode vs Commandline mode, if the input is empty
+(besides potentially the prefix character), we should probably have some sort of
+placeholder text visible to indicate which mode the user is in. Something like
+_"Enter a command name..."_ for action mode, or _"Type a wt commandline..."_ for
+commandline mode.
+
+Initially, I favored having the palette in Action Mode by default, and typing a
+`:` prefix to enter Commandline Mode. This is fairly similar to how tmux's
+internal command prompt works, which is bound to `<prefix>-:` by default.
+
+If we wanted to remain _similar_ to VsCode, we'd have no prefix character be the
+Commandline Mode, and `>` would enter the Action mode. I'd think that might
+actually be _backwards_ from what I'd expect, with `>` being the default
+character for the end of the default `cmd` `%PROMPT%`.
+
+As always, I'm also on board with the "this should be configurable by the user"
+route, so they can change what mode the command palette is in by default, and
+what the prefixes for differen modes are, but I'm not sure how we'd define that
+cleanly in the settings.
+
+```json
+{
+  "commandPaletteActionModePrefix": "", // or null, for no prefix
+  "commandPaletteCommandlineModePrefix": ">"
+}
+```
+
+We'd need to have validation on that though, what if both of them were set to
+`null`? One of them would _need_ to be `null`, so if both have a character, do
+we just assume one is the default?
+
+This counts as a future-considerations topic, and not something I'm seriously
+considering for Terminal 2.0.

```

Comment:
Just to be sure, you're saying that the "future considerations topic" is the configurability aspect, right? Mind just moving this entire section there? It's just cleaner IMO.

Reply:
No problem, it's done